samples/streaming_sample/sample_stream.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	openapi "github.com/deepfence/golang_deepfence_sdk/client"
+	oahttp "github.com/deepfence/golang_deepfence_sdk/utils/http"
+	"github.com/rs/zerolog/log"
+)
+
+var sampleApiClient *oahttp.OpenapiHttpClient
+var ctxCancelFunc context.CancelFunc
+
+func SampleStreamer(console_host, console_port,
+	console_api_token string) {
+
+	//This sampleApiClient is required to make the api calls like
+	//start secret scan, start malware scan, start vulnerability scan etc
+	sampleApiClient = oahttp.NewHttpsConsoleClient(console_host, console_port)
+	err := sampleApiClient.APITokenAuthenticate(console_api_token)
+	if err != nil {
+		log.Info().Msgf("API Client Autentication failed with error: %v", err)
+		return
+	}
+
+	streamConfig := oahttp.StreamingConfig{}
+	streamConfig.Addition = true
+	streamConfig.Deletion = true
+	streamConfig.EntityTypes = []string{"Node",
+		"Container",
+		"ContainerImage"}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	ctxCancelFunc = cancelFunc
+
+	err = oahttp.StartStreamingDelta(ctx, streamConfig, console_host,
+		console_port, console_api_token, StreamConsumerFunc, OnErrorCallback)
```

Comment:
nit: having closures instead of having `StreamConsumerFunc` and `OnErrorCallback` would look more flexible